app.py

```diff
@@ -120,8 +121,11 @@ def valid_file_ext(filename):
 
 # Runs ML on given image
 def run_ml(img_path):
-    img_file = {'image': open(img_path, 'rb')}
-    r = requests.post(url=ml_endpoint, files=img_file)
+    mime_type = mimetypes.guess_type(img_path)[0]
+    img_file = open(img_path, 'rb')
```

Comment:
If you use `with` it'll be closed at the end of the block. See the last paragraph of https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects

```
    with open(img_path, 'rb') as img_file:
        file_form = {'image': (img_path, img_file, mime_type)}

     r = requests.post(url=ml_endpoint, files=file_form)
     cap_json = r.json()
```